src/main/java/net/minecraftforge/registries/DeferredRegister.java

```diff
@@ -118,15 +162,110 @@ else if (this.superType != null)
      * @param name  Path of the registry's {@link ResourceLocation}
      * @param sup   Supplier of the RegistryBuilder that is called to fill {@link #type} during the NewRegistry event
      * @return      A supplier of the {@link IForgeRegistry} created by the builder.
+     *
+     * @deprecated Use {@link #create(ResourceLocation, String)} and {@link #makeRegistry(Class, Supplier)} instead
      */
+    @Deprecated(forRemoval = true, since = "1.18.2")
     public Supplier<IForgeRegistry<T>> makeRegistry(final String name, final Supplier<RegistryBuilder<T>> sup) {
-        if (this.superType == null)
-            throw new IllegalStateException("Cannot create a registry without specifying a base type");
-        if (this.type != null || this.registryFactory != null)
-            throw new IllegalStateException("Cannot create a registry for a type that already exists");
+        return makeRegistry(new ResourceLocation(modid, name), this.superType, sup);
+    }
+
+    /**
+     * Only used for custom registries to fill the forge registry held in this DeferredRegister.
+     *
+     * Calls {@link RegistryBuilder#setName} and {@link RegistryBuilder#setType} automatically.
+     *
+     * @param base The base type to use in the created {@link IForgeRegistry}
+     * @param sup Supplier of a RegistryBuilder that initializes a {@link IForgeRegistry} during the {@link RegistryEvent.NewRegistry} event
+     * @return A supplier of the {@link IForgeRegistry} created by the builder.
+     * Will always return null until after the {@link RegistryEvent.NewRegistry} event fires.
+     */
+    public Supplier<IForgeRegistry<T>> makeRegistry(final Class<T> base, final Supplier<RegistryBuilder<T>> sup) {
+        return makeRegistry(this.registryName, base, sup);
+    }
+
+    /**
+     * Creates a tag key based on the current modid and provided path as the location and the registry name linked to this DeferredRegister.
+     * To control the namespace, use {@link #createTagKey(ResourceLocation)}.
+     *
+     * @see #createTagKey(ResourceLocation)
+     * @see #createOptionalTagKey(String, Set)
+     */
+    @NotNull
+    public TagKey<T> createTagKey(@NotNull String path)
+    {
+        Objects.requireNonNull(path);
+        return createTagKey(new ResourceLocation(this.modid, path));
+    }
+
+    /**
+     * Creates a tag key based on the provided resource location and the registry name linked to this DeferredRegister.
+     * To use the current modid as the namespace, use {@link #createTagKey(String)}.
+     *
+     * @see #createTagKey(String)
+     * @see #createOptionalTagKey(ResourceLocation, Set)
+     */
+    @NotNull
+    public TagKey<T> createTagKey(@NotNull ResourceLocation location)
+    {
+        if (this.registryName == null)
+            throw new IllegalStateException("The registry name was not set, cannot create a tag key");
```

Comment:
The conditions of when a `IllegalStateException` is thrown should be documented.